storage/new_bucket/main.tf

```diff
@@ -32,14 +32,12 @@ resource "google_storage_bucket" "static" {
 # [END storage_create_new_bucket_tf]
 
 # [START storage_upload_object_tf]
-# Upload files
-# Discussion about using tf to upload a large number of objects
-# https://stackoverflow.com/questions/68455132/terraform-copy-multiple-files-to-bucket-at-the-same-time-bucket-creation
+# Creates a text file based on content_type with content
+# in Google Storage Bucket
 
-# The text object in Cloud Storage
 resource "google_storage_bucket_object" "default" {
   name = "new-object"
-  # Uncomment and add valid path to an object.
+  # Uncomment the following to upload an existing object from local file system
```

Comment:
I would change this to:

```
# Uncomment the following lines to upload an existing file from a 
# local file system as an object, where the content_type field must 
# match the content type of the file.

Reply:
I would simplify to
```
# To upload a local file, uncomment the following `source` and update `content_type`
```

Reply:
I've updated this. 

We have conflicting information in https://cloud.google.com/storage/docs/metadata#content-type 

```
All objects have a value specified in their Content-Type metadata, but this value does not have to match the underlying type of the object. 
```

Reply:
Nice catch. I'm happy to go with Sampath's suggestion.